noderesourcetopology: overhaul Topology Manager configuration management (#488)
kubernetes-sigs/scheduler-plugins: pkg/noderesourcetopology/README.md

using [affinity](https://kubernetes.io/docs/user-guide/node-selection/#node-affinity-beta-feature) or also
[taints](https://kubernetes.io/docs/user-guide/node-selection/#taints-and-toleations-beta-feature).

#### Topology Manager configuration

***Target audience: developers and operators of topology updaters (NodeResourceTopology producers)***

Additionally to properly partition logically your cluster like explained above, to work correctly the Topology-aware scheduler needs to know key node-specific configuration settings for the topology manager
running on the worker nodes.
This data is expected to be provided as top-level `Attributes` of the NodeResourceTopology objects:

NodeResourceTopology producers should add top-level `Attributes` in the following format
- For `Name` and `Value` of attributes, words should be `snakeCase`
- The `Name` of each attribute should be **the same of the corresponding kubelet configuration option**.
  - example: `--topology-manager-scope` becomes `topologyManagerScope`

>   - example: `topologyManagerPolicy` becomes `topologyMaanagerPolicy`

Reviewer: and I found the typo
```suggestion
  - example: `topologyManagerPolicy` becomes `topologyManagerPolicy`
```

Author: good catch, thanks! fixed

using [affinity](https://kubernetes.io/docs/user-guide/node-selection/#node-affinity-beta-feature) or also
[taints](https://kubernetes.io/docs/user-guide/node-selection/#taints-and-toleations-beta-feature).

#### Topology Manager configuration

***Target audience: developers and operators of topology updaters (NodeResourceTopology producers)***

> Additionally to properly partition logically your cluster like explained above, to work correctly the Topology-aware scheduler needs to know key node-specific configuration settings for the topology manager
> running on the worker nodes.

Reviewer 2: Just a small rephrase: 
```suggestion
In addition to logically partitioning a cluster like explained above, Topology-aware scheduler needs to know key node-specific configuration settings like Topology manager policy and scope.
```

Author: thanks, fixed

- For `Name` and `Value` of attributes, words should be `snakeCase`
- The `Name` of each attribute should be **the same of the corresponding kubelet configuration option**.
  - example: `--topology-manager-scope` becomes `topologyManagerScope`
  - example: `topologyManagerPolicy` becomes `topologyManagerPolicy`
- The `Value` of each attribute should be **one of the value of the corresponding kubelet configuration option, VERBATIM**.
  - example: `single-numa-node` becomes `single-numa-node`
- Should `topologyManagerOptions` be exposed:
  - they should be expanded in key-value pairs, using the `String()` representation
  - each key-value pair should be preceded by the `topologyManagerOption` prefix
  - every other provision described above applies
  - example: the `prefer-closest-numa-nodes` option becomes `topologyManagerOptionPreferClosestNumaNodes`, accepting exactly one of either `true` and `false`.
  - **RATIONALE**: this representation wants to guarantee all the Attribute Names are unique (no aliasing). It must be noted this is a stricter requirement with respect to the Attribute representation

>     in NRT objects, and this requirement could be lifted in the future (a upgrade path will be provided).

Reviewer 2: NIT:
``` suggestion
    in NRT objects, and this requirement could be lifted in the future (an upgrade path will be provided).
```

Author: thanks, fixed
